lumicks/pylake/force_calibration/power_spectrum.py

```diff
@@ -53,6 +53,27 @@ def downsampled_by(self, factor, reduce=np.mean):
 
         return ba
 
+    def _exclude_range(self, excluded_ranges):
+        """Exclude given frequency ranges from the power spectrum.
+
+        This function can be used to exclude noise peaks.
+
+        Parameters
+        ----------
+        exclude_ranges - `list` [`tuple` (float, float)]
+            List of ranges to exclude specified as a list of (frequency_min, frequency_max)."""
+        if not excluded_ranges:
+            return deepcopy(self)
```

Comment:
Is a `deepcopy` needed here? Wouldn't a shallow copy be enough? It feels like this would create a copy every time for the most common case of no excluded ranges.

Reply:
Good point, I'll make it a regular `copy`. It does seem overly pessimistic to always `deepcopy` it. If it was public API, there'd be more of a case for it, but it's not.